gwsumm/archive.py

```diff
@@ -77,11 +77,12 @@ def write_data_archive(outfile, channels=True, timeseries=True,
         include `EventTable` data in archive
     """
 
-    # backup existing file, in case something fails
-    backup = backup_existing_archive(outfile)
-
     try:
-        with File(outfile, 'w') as h5file:
+        # create a temporary file, this avoids overwrite the existing backup
+        temp_outfile = tempfile.mktemp(
```

Comment:
Suggestion to use `tempfile.mkstemp()` here since `mktemp()` is deprecated since version 2.3

Reply:
It's might be better/safer to use `with tempfile.NamedTemporaryFile(...)` here.

Reply:
Ok, thank you! I will implement Duncan's suggestion.

Reply:
@duncanmmacleod Side question: wouldn't either suggestion work since they are both equally "safe" now?